### ISSUE TITLE
Cornerstone events not firing when importing from NPM

### ISSUE THREAD

**Issue by dannyrb** [chafey/cornerstone#135]
Until recently, I had been including cornerstone in my project as a collection of static assets. I've recently updated my code to pull in each library as an NPM package to make it easier to manage my dependencies. I'm using `WebPack 3` + `Vue`, and I'm pulling in the following:

```
import $ from 'jquery';
import * as cornerstone from 'cornerstone-core'
import * as cornerstoneMath from 'cornerstone-math';
import * as cornerstoneTools from 'cornerstone-tools';
import * as cornerstoneWebImageLoader from 'cornerstone-web-image-loader/dist/cornerstoneWebImageLoader.js';
```

I can't seem to get any events to emit/trigger from cornerstone. However, all other functionality appears to be working. The only thing I've changed is how I pull in my libraries/dependencies.

I've tried the following to see if the event is ever fired:

*Listening for the event at the container level:*
```
document.querySelector('.dicomImageContainer').addEventListener('CornerstoneImageRendered', function(event){ console.log(event); });
$("div.dicomImage").on("CornerstoneImageRendered", this.onImageRendered);
```
*Watching for event at body:*
`monitorEvents(document.body, 'CornerstoneImageRendered')` (chrome console trick)

**Comment by swederik** [comment]
That is bizarre... I have done the same in the OHIF/Viewers Meteor app (https://github.com/OHIF/Viewers/blob/master/Packages/ohif-cornerstone/package.js) and it seems to work fine.

We changed the events that fired on $(cornerstone) to $(cornerstone.events) here recently (https://github.com/chafey/cornerstone/commit/3c3c59ec840efe28546e45016221043d97292b5b)
 but nothing has changed for events that fire on DOM elements. 

I'm at a loss at the moment as to what could be the issue. I'll try to find time to debug this, though.

**Comment by dannyrb** [comment]
Thanks, @swederik! If there's anything I can do to assist, please let me know. 

For the time being, I'm updating my zoom and wwwc values by running a method against an interval that grabs the updated values.

**Comment by leonardorame** [comment]
It looks like cornerstoneWebImageLoader is using cornerstone.events, while cornerstoneWADOImageLoader is still using the old method.

**Comment by leonardorame** [comment]
I can confirm this, @swederik  could you replace this:

        $(cornerstone).trigger('CornerstoneImageLoadProgress', {

with this:

        $(cornerstone.events).trigger('CornerstoneImageLoadProgress', {

in cornerstoneWADOImageLoader?.

**Comment by dannyrb** [comment]
@swederik finally figured this one out.

I'm using Webpack & VueJs. The proposed solution for including jQuery globally is adding something similar to the following to plugins:

```
    new webpack.ProvidePlugin({
        $: "jquery",
        jQuery: "jquery"
    })
```

I was under the impression that this was working because:

- The viewer would load, `cornerstoneTools` would function, all was good in the world
- Typing `$` in the console would show that jQuery had been loaded

In reality, the plugin provided jQuery in each instance it was needed, but not globally. This allowed everything to build, but I noticed a few wonky issues:

- Certain `cornerstoneTools` would not activate/work (length, probe, highlight, etc).
- Events for `CornerstoneImageRendered` would not fire.

As it turns out, I did not have jQuery added globally. ChromeDevTools provides a lightweight instance of `$` in the console to make it easier to select DOM elements.

Removing the webpack.ProvidePlugin code and adding the following to my viewer component resolved all issues:

```
import $ from 'jquery'; // 2.2.4 (matching cornerstoneTools to be safe)
global.$ = require('jquery');
```

JS isn't my first language, so I'm sure there's a more eloquent explanation. The above worked for me -- Not a bug with `cornerstone`, but it would be nice if jQuery did not need to be available globally for `cornerstone` and `cornerstoneTools` to work.
